rubix-hadoop2/src/main/java/com/qubole/rubix/hadoop2/CachingDistributedFileSystem.java

```diff
@@ -32,25 +34,17 @@
   @Override
   public void initialize(URI uri, Configuration conf) throws IOException
   {
-    LOG.debug("Initializing DistributedFileSystem - Hadoop2");
-    if (clusterManager == null) {
-      initializeClusterManager(conf);
+    try {
+      initializeClusterManager(conf, ClusterType.HADOOP2_CLUSTER_MANAGER);
+      super.initialize(uri, conf);
+    }
+    catch (ClusterManagerInitilizationException ex) {
+      throw new IOException(ex);
     }
-    setClusterManager(clusterManager);
-    super.initialize(uri, conf);
   }
 
   public String getScheme()
   {
     return SCHEME;
   }
-
-  private synchronized void initializeClusterManager(Configuration conf)
-  {
-    if (clusterManager != null) {
-      return;
-    }
-    clusterManager = new Hadoop2ClusterManager();
```

Comment:
All these classes have a local copy of the derived class of cluster manager and you are not setting those anymore. Definitely needs to be fixed in all the classes.

Reply:
this clustermanager is variable is defined in base class(CachingFileSystem). I am calling setClusterManager in the initializeClusterManager method in the base class
https://github.com/qubole/rubix/pull/156/files/b97f3d8fa3dbbba754a9a564b38520142ccb841b#diff-a33cf3db53caa949ee05fad1d36c82f2R121

Reply:
There is yet another in this class and other derived classes of `CachingFileSystem`. Check line number 31(new) or 29(old)

Reply:
Or you can remove the one in all the derived classes. They seem unnecessary.

Reply:
Done

Reply:
Done